polaris-react/src/components/OptionList/OptionList.tsx

```diff
@@ -39,6 +39,10 @@ export interface OptionListProps {
   verticalAlign?: Alignment;
   /** Callback when selection is changed */
   onChange(selected: string[]): void;
+  /** Callback when pointer enters an option */
+  onPointerEnterOption?(selected: string): void;
+  /** Callback when focusing an option */
+  onFocusOption?(selected: string): void;
```

Comment:
```suggestion
  onOptionPointerEnter?(selected: string): void;
  /** Callback when focusing an option */
  onOptionFocus?(selected: string): void;
```
Wondering if this reads a little better. Not a strong opinion here

Reply:
I think `onPointerEnterOption` and `onFocusOption` make sense here, the other Polaris components have an `onVerbNoun` naming convention for events triggered by user action 💯